### PR TITLE
Update github status on build update/stop

### PR DESCRIPTION
**Pull request by d2lam**
Update github status on build update and build stop

### CODE REVIEW

File: lib/base.js [hunk 1]
```diff
@@ -80,6 +76,20 @@ class BaseModel {
     }
 
     /**
+     * Check if the model data is dirty.
+     * If key is passed in, check if the key is dirty.
+     * @param  {String}  [key]    They key to check
+     * @return {Boolean}
+     */
+    isDirty(key) {
```
Comment (petey) [inline lib/base.js]: Please define in source order before used.

File: test/lib/build.test.js [hunk 2]
```diff
@@ -149,15 +153,152 @@ describe('Build Model', () => {
...
+            });
+
+            githubMock.getInfo.returns({
+                user: 'screwdriver-cd',
+                repo: 'models'
+            });
+            githubMock.run.resolves(null);
+        });
+
+        it('promises to stop a build when it is running', () =>
```
Comment (petey) [inline test/lib/build.test.js]: you mean queued?

**Comment by petey** [comment]
👍

File: lib/build.js [hunk 3]
```diff
@@ -183,12 +183,63 @@ class BuildModel extends BaseModel {
...
+        const dirty = this.isDirty('status');
+
+        return super.update()
+            .then(() => {
+                if (!dirty) {
+                    return this;
+                }
+
+                return this.pipeline
+                    // stop the build
```
Comment (nkatzman) [inline lib/build.js]: is this comment accurate?

Comment (petey) [inline lib/build.js]: copy/paste strikes again!

Comment (d2lam) [inline lib/build.js]: good catch

**Comment by nkatzman** [comment]
Besides my small comments, 👍

**Pull request merged**